### PR TITLE
lat-lon endpoints

### PR DESCRIPTION
**Pull request by matthewleemle**
Features:
1. changed secret to be sealed with the user id instead of the username
2. included endpoints for updating user location, endpoint: /users/:id/location, JSON format as following: 
{
  "lat": Number,
  "lon": Number
}

### CODE REVIEW

File: Controllers/user.js [hunk 1]
```diff
@@ -224,6 +224,17 @@ async function toggleStatus(req,res){
   }
 }
 
+async function addLocation(req,res){
+  var query = {_id: new ObjectId(req.params.id)};
+  try{
+    var result = await UserModel.findOneAndUpdate(query, {location: {lat: req.body.lat, lon: req.body.lon}}, {useFindAndModify: false});
```
Comment (Asiapenguin) [inline Controllers/user.js]: What does useFindAndModify do here?

Comment (matthewleemle) [reply Controllers/user.js]: findAndModify is a mongoDB method, and mongoose gives a deprecation warning if I don't include {useFindAndModify: false} as an option

Comment (Asiapenguin) [reply Controllers/user.js]: Can you set it globally with
`mongoose.set('useFindAndModify', false);`

Comment (matthewleemle) [reply Controllers/user.js]: I don't think so; {useFindAndModify: false} is an object used as a parameter (labeled as options in the Mongoose API, you can look it up) for the findOneAndUpdate function (which is a method for Models in Mongoose, you can look it up). Feel free to look up the Mongoose API docs for clarifications

Comment (Asiapenguin) [reply Controllers/user.js]: https://mongoosejs.com/docs/deprecations.html#findandmodify

Comment (matthewleemle) [reply Controllers/user.js]: ahhhhhhhhh I see, thanks

Comment (matthewleemle) [reply Controllers/user.js]: check the new commits, fixed it

Comment (matthewleemle) [reply Controllers/user.js]: check the new commits, fixed it

**Pull request merged**